src/main/java/org/broadinstitute/hellbender/tools/ConvertHadoopBamFragmentToBam.java

```diff
@@ -0,0 +1,88 @@
+package org.broadinstitute.hellbender.tools;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.BinaryCodec;
+import htsjdk.samtools.util.BlockCompressedOutputStream;
+import htsjdk.samtools.util.BlockCompressedStreamConstants;
+import htsjdk.samtools.util.RuntimeIOException;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgram;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.TestSparkProgramGroup;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+
+import java.io.*;
+
+
+/**
+ * Spark troubleshooting utility that converts a Hadoop bam fragment (eg., a part0000, part0001, etc. file
+ * produced by ReadsSparkSink) into a readable bam file by adding a header and a BGZF terminator.
```

Comment:
This should be a `{@link ReadSparkSink}` so it doesn't get forgotten on a rename.